grpc/cargo_casper/tests/integration_tests.rs

```diff
@@ -71,6 +71,9 @@ fn run_tool_and_resulting_tests(turbo: bool) {
     test_cmd.arg("test").current_dir(test_dir.join("tests"));
     let test_output = output_from_command(test_cmd);
     assert_eq!(SUCCESS_EXIT_CODE, test_output.status.code().unwrap());
+
+    // Cleans up temporary directory, but leaves it otherwise if the test failed.
```

Comment:
Is that correct?  I thought `tempdir` deleted any directories created when dropped.

Reply:
It deletes everything when dropped, but you can make it not do so via `into_path`.

Reply:
I'm in favor of always removing thing, to be honest. Is there a reason we ever want the tempdir to be lying around?

Reply:
No - the existing behaviour was an oversight on my part.

My point is that I think this comment is invalid, and also that the remove_dir calls are unneeded.

Reply:
> 
> 
> I'm in favor of always removing thing, to be honest. Is there a reason we ever want the tempdir to be lying around?

In case of tmp build failures it gives you option to manually inspect cargo project and rerun build and see what fails. It's hard debug failure otherwise

Reply:
> My point is that I think this comment is invalid, and also that the remove_dir calls are unneeded.

It's valid though - I'm calling `into_path` at the beginning which bypasses TempDir Drop impl completely

Reply:
I didn't actually think anyone would ever do that! :D

Normally the failure can be replicated by building the crate which failed on its own.  I.e. if the reported error was in building `casper_node`, I'd just cd to `node` and build from there.

But I can see your point - if it's some spurious failure which can't be easily replicated, maybe it _is_ better to keep the temp dir around.

Reply:
> 
> 
> I didn't actually think anyone would ever do that! :D
> 
> Normally the failure can be replicated by building the crate which failed on its own. I.e. if the reported error was in building `casper_node`, I'd just cd to `node` and build from there.
> 
> But I can see your point - if it's some spurious failure which can't be easily replicated, maybe it _is_ better to keep the temp dir around.

Thanks. Sorry I meant specifically that from time to time I have to cd into /tmp/.tmpFooBar and run build from there, and also modify the code there to fix it, then copy and paste it into the boilerplate, and then `make check` again

Reply:
I've never done that!